Common/Data/Custom/SmartInsider/SmartInsiderIntention.cs

```diff
@@ -0,0 +1,339 @@
+                NextResultsAnnouncementsDate = NextResultsAnnouncementsDate,
+                NextCloseBegin = NextCloseBegin,
+                LastCloseEnded = LastCloseEnded,
+                SecurityDescription = SecurityDescription,
+                TickerCountry = TickerCountry,
+                TickerSymbol = TickerSymbol,
+                AnnouncementDate = AnnouncementDate,
+                TimeReleased = TimeReleased,
+                TimeProcessed = TimeProcessed,
+                TimeReleasedUtc = TimeReleasedUtc,
+                TimeProcessedUtc = TimeProcessedUtc,
+                AnnouncedIn = AnnouncedIn,
+
+                IntentionVia = IntentionVia,
+                IntentionBy = IntentionBy,
+                BuybackIntentionHoldingType = BuybackIntentionHoldingType,
+                IntentionAmount = IntentionAmount,
+                ValueCurrency = ValueCurrency,
+                IntentionValue = IntentionValue,
+                IntentionPercentage = IntentionPercentage,
+                IntentionAuthorisationStartDate = IntentionAuthorisationStartDate,
+                IntentionAuthorisationEndDate = IntentionAuthorisationEndDate,
+                PriceCurrency = PriceCurrency,
+                MinimumPrice = MinimumPrice,
+                MaximumPrice = MaximumPrice,
+                BuybackIntentionNoteText = BuybackIntentionNoteText
+            };
+        }
+
+        /// <summary>
+        /// Converts the data to CSV
+        /// </summary>
+        /// <returns>String of CSV</returns>
+        /// <remarks>Parsable by the constructor should you need to recreate the object from CSV</remarks>
+        public override string ToCsv()
+        {
+            return $"{TransactionID}\t{BuybackType}\t{LastUpdate:yyyyMMdd}\t{LastIDsUpdate:yyyyMMdd}\t{ISIN}\t{USDMarketCap}\t{CompanyID}\t{ICBIndustry}\t{ICBSuperSector}\t{ICBSector}\t{ICBSubSector}\t{ICBCode}\t{CompanyName}\t{PreviousResultsAnnouncementDate:yyyyMMdd}\t{NextResultsAnnouncementsDate:yyyyMMdd}\t{NextCloseBegin:yyyyMMdd}\t{LastCloseEnded:yyyyMMdd}\t{SecurityDescription}\t{TickerCountry}\t{TickerSymbol}\t{AnnouncementDate:yyyyMMdd}\t{TimeReleased:yyyyMMdd HH:mm:ss}\t{TimeProcessed:yyyyMMdd HH:mm:ss}\t{TimeReleasedUtc:yyyyMMdd HH:mm:ss}\t{TimeProcessedUtc:yyyyMMdd HH:mm:ss}\t{AnnouncedIn}\t{IntentionVia}\t{IntentionBy}\t{BuybackIntentionHoldingType}\t{IntentionAmount}\t{ValueCurrency}\t{IntentionValue}\t{IntentionPercentage}\t{IntentionAuthorisationStartDate:yyyyMMdd}\t{IntentionAuthorisationEndDate:yyyyMMdd}\t{PriceCurrency}\t{MinimumPrice}\t{MaximumPrice}\t{BuybackIntentionNoteText}";
+        }
+
+        /// <summary>
+        /// Determines equality to another SmartInsiderIntention instance
+        /// </summary>
+        /// <param name="other">Another SmartInsiderIntention instance</param>
+        /// <returns>Boolean value indicating equality</returns>
+        public override bool Equals(SmartInsiderEvent other)
```

Comment:
If overriding the `Equals` method, standard practice is to _also_ override the `GetHashCode` method. R# provides a code generator for both of these. Place cursor on the type. Go to `ReSharper -> Edit -> Generate Code -> Equality Members`